### ISSUE TITLE
Use HTTPS (TLS/SSL) exclusively

### ISSUE THREAD

**Issue by achadwick** [mypaint/mypaint.github.io#24]
Would it be possible to make the Discourse site use `https://` only? Obtaining a certificate appears to be easy and free these days: <https://letsencrypt.org/>. I don't know the hosting arrangements though - there are a number of other requirements for this such as using a dedicated IP and configuring nginx appropriately :smile:

The traditional way of making old HTTP links is by, once the HTTPS site is running in parallel, sending like-for-like `301 Moved Permanently` or `302 Found` responses from `http://site/foo` to `https://site/foo`.

I want to avoid users having to send passwords in the clear, so this would work in conjunction with #19 for user security purposes.

**Comment by odysseywestra** [comment]
Discourse says they are working on and easy way to enable HTTPS with Let's Encrypt. There is a way to enable it manually, but I want to wait till Discourse comes up with a solution.  Here's a thread about on their Meta Site talking about it: 

https://meta.discourse.org/t/support-for-lets-encrypt/22308/28 

I would love to get it enabled for both our Community Forums and our main site as well, but for now it's a waiting game.

**Comment by achadwick** [comment]
The wait might be a long one. Ordinarily, I'd say that if it can be done manually, it should be done manually. This is too big a hole to leave open, and the wait might be a long one.

However there's that 90 day renewal requirement https://letsencrypt.readthedocs.org/en/latest/using.html#renewal and the port 80 binding thing that the letsencrypt tools need. I wouldn't feel super happy automating something to bounce discourse out of the way every 3 months just to renew its cert.

For now, let's concentrate on shoring up OAuth ☺

**Comment by odysseywestra** [comment]
Yeah, it could be a long wait. We could also move our site to our Discourse Server using [Jekyll-hook](https://github.com/developmentseed/jekyll-hook). Our Server is Debian with Docker on top. So we could just create another container dedicated for our website and host it there. That way we don't have to wait on Github Pages to enable HTTPS for custom domains, plus we have more control of our site. There are a few hurdles to accomplish this, but I think I can do it.

**Comment by odysseywestra** [comment]
Found this script on the Discourse Meta site which will do the auto-renew for us. So I would probably need to create a dedicated email for MyPaint. @achadwick Do we have a dedicated email for MyPaint?

https://meta.discourse.org/t/setting-up-lets-encrypt-cert-with-discourse-docker/40709

**Comment by achadwick** [comment]
@odysseywestra Sure - I'll get one set up with Mythic-beasts if all that is needed is a redirect to a different email address. It'll be specifically a Let's Encrypt one so we can filter it.

Shall I redirect it to just yourself, or all of the core devs, or some other arrangement?

**Comment by odysseywestra** [comment]
@achadwick sure if you want to go that route. We could also create a dedicated google account for MyPaint as well and set up it's gmail address as a forwarder too. The reason I'm suggesting that is that I would like to switch to Google's SMTP Mail Delivery system from Mailguns. That way we could take advantage of reply by email for the forums.

**Comment by achadwick** [comment]
@odysseywestra Maybe I don't understand the requirement for Let's Encrypt. Please can you tell me what you need the email address for? Does email sent to it need to be processed automatically, or is it just an address that exists so that Let's Encrypt can contact one of us?

Processing replies by email seems off-topic for this issue. We already have the ability to forward email from an `@mypaint.org` address to a small, managed set of addresses, if you want to get SSL up and running first and the use case is that of a non-personal contact address.

**Comment by odysseywestra** [comment]
According to the thread, the email is used as a notification system to in case the renewal goes wrong. So yeah go ahead and create an admin@mypaint.org or developers@mypaint.org, and at least have it forward to my email. It's my username at gmail.com. We can talk about the whole google account thing in another issue thread.

**Comment by odysseywestra** [comment]
Okay I tired to enable HTTPS with that script mentioned above. After four rebuilds I kept getting errors with let's encrypt so I'll try enable it next time after I talk to someone about over on Discourses Meta Forums.

**Comment by odysseywestra** [comment]
Okay, HTTPS support is now enable on our Community Forums.
